piff/hyperpsf.py

```diff
@@ -0,0 +1,277 @@
+        if self._interp_init:
+            for i, val in enumerate(self._interp_init):
+                self._fit_func_kwargs_minuit['p{0}'.format(i + Nmodel)] = val
+        # set rest
+        for i in xrange(Nkeys, 50):
+            self._fit_func_kwargs_minuit['p{0}'.format(i)] = 0
+
+        # set initial step sizes
+        if self._model_error:
+            for i, val in enumerate(self._model_error):
+                self._fit_func_kwargs_minuit['error_p{0}'.format(i)] = val
+        if self._interp_error:
+            for i, val in enumerate(self._interp_error):
+                self._fit_func_kwargs_minuit['error_p{0}'.format(i + Nmodel)] = val
+
+        # set limits
+        if self._model_limit:
+            for i, val in enumerate(self._model_limit):
+                self._fit_func_kwargs_minuit['limit_p{0}'.format(i)] = val
+        if self._interp_limit:
+            for i, val in enumerate(self._interp_limit):
+                self._fit_func_kwargs_minuit['limit_p{0}'.format(i + Nmodel)] = val
+
+        # run initial solves for model and interp
+        self.model.solve(self._stars)
+        self.interp.solve(self._stars)
+
+    # forgive me oh lord this python sin
+    def _fit_func(self, p0, p1, p2, p3, p4, p5, p6, p7, p8, p9,
+                  p10, p11, p12, p13, p14, p15, p16, p17, p18, p19,
+                  p20, p21, p22, p23, p24, p25, p26, p27, p28, p29,
+                  p30, p31, p32, p33, p34, p35, p36, p37, p38, p39,
+                  p40, p41, p42, p43, p44, p45, p46, p47, p48, p49,
+                  ):
+        # convert p to params
+        params = [p0, p1, p2, p3, p4, p5, p6, p7, p8, p9,
+                  p10, p11, p12, p13, p14, p15, p16, p17, p18, p19,
+                  p20, p21, p22, p23, p24, p25, p26, p27, p28, p29,
+                  p30, p31, p32, p33, p34, p35, p36, p37, p38, p39,
+                  p40, p41, p42, p43, p44, p45, p46, p47, p48, p49,]
+
+        return self._fit_func_interior(params)
+
+    def _fit_func_interior(self, params):
+        # update model and interp via params
+        if len(self._model_keys) > 0:
+            model_params = params[:len(self._model_keys)]
+            model = {self._model_keys[i]: model_params[i]
+                     for i in xrange(len(self._model_keys))}
+            self.model.update(logger=self._logger, **model)
+            self.model.solve(self._stars)
+
+        if len(self._interp_keys) > 0:
+            interp_params = params[len(self._model_keys):len(self._model_keys) + len(self._interp_keys)]
+            interp = {self._interp_keys[i]: interp_params[i]
+                     for i in xrange(len(self._interp_keys))}
+            self.interp.update(logger=self._logger, **interp)
+            self.interp.solve(self._stars)
+
+        # interp
+        stars_interp = self.interp.interpolateList(self._stars, logger=self._logger)
+
+        # model
+        stars_model = [self.model.draw(star) for star in stars_interp]
```

Comment:
No problem with adding a `drawList` function analogous to `interpolateList` for this.